### PR TITLE
feat(backend): Initialized Go backend service and added basic structure

### PR DESCRIPTION
**Pull request by ivantomic77**
Added Go backend basic structure with one /api/health endpoint.

### CODE REVIEW

File: backend/internal/handlers.go
Comment (LukaPlejic) [inline backend/internal/handlers.go:1]: Rename this file to `health.go`
and change it's path to `backend\internal\handlers\health.go`

File: backend/cmd/api/api.go
Comment (LukaPlejic) [inline backend/cmd/api/api.go:1]: Move this file to `backend\internal\api`

File: backend/cmd/api/main.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	log.Fatal(app.run(app.mount()))
+}
+
+func getEnvOrDefault(envVar, defaultVal string) string {
+	if value := os.Getenv(envVar); value != "" {
+		return value
+	}
+	return defaultVal
+}
```
Comment (LukaPlejic) [inline backend/cmd/api/main.go]: This can be extracted to a specific `helpers` package for further reusability

File: backend/cmd/api/api.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/FlamingoTP/cardz/internal"
+)
+
+type application struct {
+	config config
+}
+
+type config struct {
+	addr string
+}
```
Comment (LukaPlejic) [inline backend/cmd/api/api.go]: Use Uppercase for Struct names
e.g. `Application`, `Config`

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
+= CardZ Project
+:toc: auto
+
+CardZ is an open-source platform for creating card decks and enabling users to play them in various game modes. The platform encourages self-hosting, giving users full control over their experience.
+
+== Features
+
+Coming soon
+
+== Changelog
```
Comment (LukaPlejic) [inline README.adoc:10]: Shouldn't changelog be it's own .md file?

**Review by LukaPlejic — CHANGES_REQUESTED** [review]

File: README.adoc (hunk 3 above)
Comment (ivantomic77) [reply README.adoc:10]: Yes, it will be. I left a slot for the link to the CHANGELOG file.

**Comment by ivantomic77** [comment]
@LukaPlejic Please check the changes I made and close threads if resolved.

**Review by LukaPlejic — APPROVED** [review]

**Pull request merged**